contrib/src/test/java/com/datatorrent/contrib/avro/AvroReaderWriterTest.java

```diff
@@ -0,0 +1,219 @@
+
+import junit.framework.Assert;
+
+@SuppressWarnings("deprecation")
+public class AvroReaderWriterTest {
+
+	private AvroToPojo reader = null;
+	private PojoToAvro writer = null;
+
+	public static final String schemaFile = "{\"namespace\":\"abc\",\"type\":\"record\",\"doc\":\"Order schema\",\"name\":\"Order\",\"fields\":[{\"name\":\"orderId\",\"type\": \"long\"},{\"name\":\"customerId\",\"type\": \"int\"},{\"name\":\"total\",\"type\": \"double\"},{\"name\":\"customerName\",\"type\": \"string\"}]}";
+
+	public static final String fieldInfoInitMap = "orderId:orderId:LONG,customerId:customerId:INTEGER,customerName:customerName:STRING,total:total:DOUBLE";
+
+	private Class<?> className = SimpleOrder.class;
+
+	private List<GenericRecord> recordList = null;
+
+	public AvroReaderWriterTest() {
+
+	}
+
+	@Before
+	public void initializeReaderAndWriter() {
+		reader = new AvroToPojo();
+		writer = new PojoToAvro();
+
+		recordList = new LinkedList<GenericRecord>();
+
+		initializeWriter();
+		initializeReader();
+
+	}
+
+	public void initializeWriter() {
+
+		getWriter().setSchemaString(schemaFile);
+
+		try {
+			getWriter().parseSchema();
+		} catch (IOException e) {
+			LOG.error("Error in parsing");
+		}
+
+		getWriter().setCls(className);
+
+		getWriter().initializeColumnMap(writer.getSchema());
+
+	}
+
+	public void createReaderInput() {
+		int cnt = 3;
+
+		while (cnt > 0) {
+
+			GenericRecord rec = new GenericData.Record(reader.getSchema());
+			rec.put("orderId", cnt * 1);
+			rec.put("customerId", cnt * 2);
+			rec.put("total", cnt * 1.5);
+			rec.put("customerName", "*" + cnt + "*");
+			cnt--;
+			recordList.add(rec);
+
+		}
+	}
+
+	public void initializeReader() {
+		reader.setSchemaString(schemaFile);
+
+		try {
+			getReader().parseSchema();
+		} catch (IOException e) {
+			LOG.error("Exception in parsing schema," + e.getMessage());
+		}
+
+		getReader().setCls(className);
+
+		/**
+		 * To set the field mapping we use comma separated list of
+		 * fieldInPojo:FieldInGenericRecord:DataType This can be set as a part
+		 * of the operator initialization phase on the fieldInfoInitMap string
+		 * If this is not given the reflection is used only once to generate
+		 * this map.
+		 */
+
+		if (fieldInfoInitMap.isEmpty() || getReader().getGenericRecordToPOJOFieldsMapping() == null) {
+			getReader().setFieldInfos(
+					getReader().createFieldInfoMap(getReader().generateFieldInfoInputs(SimpleOrder.class)));
+		} else {
+			getReader().setFieldInfos(getReader().createFieldInfoMap(fieldInfoInitMap));
+		}
+
+		getReader().initColumnFieldSetters(getReader().getFieldInfos());
+
+		createReaderInput();
+
+	}
+
+	public AvroToPojo getReader() {
+		return reader;
+	}
+
+	public void setReader(AvroToPojo reader) {
+		this.reader = reader;
+	}
+
+	public PojoToAvro getWriter() {
+		return writer;
+	}
+
+	public void setWriter(PojoToAvro writer) {
+		this.writer = writer;
+	}
+
+	@Test
+	public void testWriter() {
```

Comment:
You need to make operator go through proper lifecycle in unit tests.
op.setup()
op.beingWindow(x)
op.port.put(<tuple>)
op.endWindow();
op.teardown()